axis.go

```diff
@@ -424,26 +429,88 @@ var _ Ticker = LogTicks{}
 // Ticks returns Ticks in a specified range
 func (LogTicks) Ticks(min, max float64) []Tick {
 	var ticks []Tick
+	var labels []float64
+	var new_labels []float64
+	labels = nil
 	val := math.Pow10(int(math.Floor(math.Log10(min))))
+
 	if min <= 0 {
 		panic("Values must be greater than 0 for a log scale.")
 	}
-	prec := precisionOf(max)
+
 	for val < max*10 {
 		for i := 1; i < 10; i++ {
 			tick := Tick{Value: val * float64(i)}
 			if i == 1 {
-				tick.Label = formatFloatTick(val*float64(i), prec)
+				labels = append(labels, val*float64(i))
 			}
+			//Makes a list of small ticks
 			ticks = append(ticks, tick)
 		}
 		val *= 10
 	}
-	tick := Tick{Value: val, Label: formatFloatTick(val, prec)}
-	ticks = append(ticks, tick)
+
+	new_labels = adjustPrecision(labels)
+
+	for j := 0; j < len(labels); j++ {
+		//Adds big ticks to the list of small ones
+		ticks = append(ticks, Tick{Value: labels[j], Label: formatFloatTick(new_labels[j], -1)})
+	}
 	return ticks
 }
 
+func adjustPrecision(elements []float64) []float64 {
+	const maxExp = 308 //precision of the MaxFloat64 which we take as the maximum one here
+	for i := 1; i < maxExp; i++ {
+		var result []float64
+		i10 := math.Pow10(i)
+
+		for _, v := range elements {
+			var new_element float64
+			vpow := v * i10
+			//As within 'if' statement vpow is turned to integer, it cannot exceed the MaxInt64. If it does, another rounding technique is used.
```

Comment:
This comment is not necesary, and it turns out also incorrect for some platforms; in the case that this code is written compiled on a 32 but arch, the guard you have will fail to prevent a truncation since `sizeof(int)` < `sizeof(int64`. This is (probably) obviated by changing to use `math.Trunc`, but I'd like to understand/know what the problem is with using the code from `floats.Round`.

Reply:
So, if I change the function to this:
`func adjustPrecision(elements []float64) []float64 {`
`        const maxExp = 308 //maxExp is the maximum float64 exponent`
`        for i := 1; i < maxExp; i++ {`
`        var result []float64`
`            for _, v := range elements {`
`                       newElement := floats.Round(v,i)`
`                       result = append(result, newElement)`
`                        }`
`                if !hasDuplicates(result) {`
`                 return result`
`                 }`
`         }`
`      return nil`
`}`
the code fails the TestAxisSmallTick (axis_test.go).

Here are the test results:
	axis_test.go:63: tick labels mismatch:
		got: ["-2e-15" "-1e-15" "0"]
		want:["-1.6e-15" "-8e-16" "0"]
	axis_test.go:63: tick labels mismatch:
		got: ["4.799999999999997e+307" "5.199999999999997e+307" "5.599999999999996e+307"]
		want:["4.8e+307" "5.2e+307" "5.6e+307"]

TestLegendAlignment fails too (cannot catch the difference visually).

My new test TestFloatPrecision fails too, but here it is fine - ticks are just a bit different (1299-1299.6-1300.2-1300.8).

Reply:
I think those failures are OK. The first one is an aesthetic choice and the second one is unreasonable to expect correctness for.

Note that while you can test with `floats.Round` you cannot import `floats` and will have to copy-paste the implementation into an unexported function here - we don't import any gonum code into plot to avoid having people who just want plotting to have to bring all the rest of the suite.

Also, you may want to look at how to use \`\`\` and \` marks for \<pre\> blocks. See [here](https://guides.github.com/features/mastering-markdown/).

Reply:
Actually `floats` have been imported to the code when I started to make changes. Never mind. I have replaced the function made by myself with `Round`. Under the same name. 

Thanks a lot for your help!